src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,398 @@
+    }
+
+    @Override
+    public Queue updateQueue(final QueueVO updateRequest) {
+        checkRegistrationRequest(updateRequest);
+
+        final String updateName = updateRequest.getName();
+        final List<String> updateHostList = updateRequest.getHostList();
+        final List<String> updateUserGroups = updateRequest.getAllowedUserGroups();
+        if (updateName == null || updateHostList == null || updateUserGroups == null) {
+            throw new UnsupportedOperationException("Name, hostList and allowedUserGroups should be specified for "
+                    + "successful partition update");
+        }
+
+        final Context context = createContextWithUpdatePartitionName(updateName);
+        final CommandResult sinfoResult = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, context));
+        checkIfExecutionResultIsEmpty(sinfoResult);
+
+        final List<List<String>> partitionData = getNodesData(sinfoResult.getStdOut());
+
+        final List<String> currentNodesDecrypted = getNodesFromPartitionDataAndDecrypt(partitionData);
+        final List<String> userGroups = getUserGroupsFromPartitionData(partitionData);
+        final int cpus = getCpusFromPartitionData(partitionData);
+
+        final List<String> updateHostListDecrypted = decryptGroupOfNodes(updateHostList);
+        sortList(updateUserGroups);
+
+        if (updateHostListDecrypted.equals(currentNodesDecrypted) && updateUserGroups.equals(userGroups)) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "New partition properties and the current one are "
+                    + "equal");
+        }
+        fillContextWithDateToUpdate(context, updateUserGroups, updateHostListDecrypted);
+
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SCONTROL_COMMAND, context));
+        checkIsResultIsCorrect(result);
+
+        return Queue.builder()
+                .name(updateName)
+                .hostList(updateHostListDecrypted)
+                .allowedUserGroups(updateUserGroups)
+                .slots(new SlotsDescription(cpus, Collections.emptyMap()))
+                .build();
+    }
+
+    @Override
+    public List<Queue> listQueues() {
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, new Context()));
+        checkIsResultIsCorrect(result);
+
+        if (result.getStdOut().size() == 0) {
+            return Collections.EMPTY_LIST;
+        }
+
+        return fillQueueNameFromOutput(result.getStdOut());
+    }
+
+    @Override
+    public List<Queue> listQueues(final QueueFilter queueFilter) {
+        final Context context = new Context();
+        if (queueFilter.getQueues() != null) {
+            context.setVariable(SCONTROL_PARTITION_NAME, String.join(COMMA, queueFilter.getQueues()));
```

Comment:
maybe just `PARTITIONS` ?

if we move `String.join` to the command template I guess the code will be more readable, and empty/null-checking too

in a template could be 
```
[# th:if="${not #lists.isEmpty(partitions)}"]
    -p
    [# th:text="${#strings.listJoin(partitions, ',')}"]
    [/]
[/]
```

there are a few other places with the same

Reply:
fixed

Reply:
we could remove `if (queueFilter.getQueues() != null)` because it is in the template now
345, 348 - the same

Reply:
if we know. that variable is null, why should we set it to the context?

Reply:
sorry, it's my fail - I mixed up with #strings.isEmpty (it checks for null)